packages/node/src/behaviors/access-control/AccessControlServer.ts

```diff
@@ -399,86 +435,168 @@ export class AccessControlServer extends AccessControlBehavior.with("Extension")
      * validation.
      */
     protected extensionEntryAccessCheck(
-        _aclList: AccessControlTypes.AccessControlEntry[],
-        _aclEntry: AccessControlTypes.AccessControlEntry,
+        _aclList: AclList,
+        _aclEntry: AclEntry,
         _subjectDesc: IncomingSubjectDescriptor,
         _endpoint: AclEndpointContext,
         _clusterId: ClusterId,
     ) {
         return true;
     }
 
+    /** A fabric was added or updated, so we need to initialize the ACL for this fabric */
+    #updateFabricAcls(fabric: Fabric) {
+        const fabricIndex = fabric.fabricIndex;
+        fabric.acl.aclList = deepCopy(this.state.acl).filter(entry => entry.fabricIndex === fabricIndex);
+    }
+
+    /**
+     * When beginning an interaction for an online session, we register the potential ACL change for the associated
+     * fabric index. If ACL data are really changed later, the exchange gets added then.
+     */
+    #handleInteractionBegin(session?: AccessControl.Session) {
+        if (session !== undefined && !session.offline && session.fabric !== undefined) {
+            this.#prepareAclUpdateFor(session.fabric);
+        }
+    }
+
     /**
-     * The AccessControlManager instance that is used to manage the ACL for this behavior.
+     * When an interaction is finished, we check if there was a delayed ACL update for the associated fabric and apply
+     * it to the manager. For this we check if we have an exchange stored because otherwise the interaction was in fact
+     * not changing the ACL.
      */
-    get aclManager() {
-        if (this.internal.aclManager === undefined) {
-            throw new InternalError("ACL manager not initialized yet");
+    #handleInteractionEnd(session?: AccessControl.Session) {
+        if (session !== undefined && !session.offline && session.fabric !== undefined) {
+            if (this.internal.aclUpdateDelayed.get(session.fabric) !== undefined) {
+                this.#applyDelayedAclUpdateFor(session.fabric);
+            }
         }
-        return this.internal.aclManager;
     }
 
-    #updateAccessControlList(acl: AccessControlTypes.AccessControlEntry[]) {
-        if (!this.aclUpdateDelayed) {
-            logger.info("ACL updated, updating ACL manager", acl);
-            this.aclManager.updateAccessControlList(deepCopy(acl));
+    /** The ACL list was changed, so we need to determine if and when to apply the update to the ACL manager */
+    #updateAccessControlList(
+        acl: AccessControlTypes.AccessControlEntry[],
+        _oldAcl: AccessControlTypes.AccessControlEntry[],
+        context?: ActionContext,
+    ) {
+        if (context === undefined || context.offline) {
+            // local or offline ACL change, so we update all fabrics because we do not know better
```

Comment:
Could we diff oldAcl and acl to identify changed fabrics?  Or is this too uncommon to worry about?